module/spl/spl-kmem.c

```diff
@@ -133,6 +133,32 @@ strfree(char *str)
 }
 EXPORT_SYMBOL(strfree);
 
+#ifndef __GFP_REPEAT
+#define	__GFP_REPEAT	__GFP_RETRY_MAYFAIL
+#endif
+void *
+spl_kvmalloc(size_t size, gfp_t flags)
+{
+	gfp_t kmalloc_flags = flags;
+	void *ret;
+#ifdef HAVE_KVMALLOC
+	if ((flags & GFP_KERNEL) == GFP_KERNEL) {
+		return (kvmalloc(size, flags));
+	}
+#endif
+	if (size > PAGE_SIZE) {
+		kmalloc_flags |= __GFP_NOWARN;
```

Comment:
this is only true if you call spl_kvmalloc with kmem_flags_convert. but you may also call it direct with own flags. kmalloc_flags | __GFP_NOWARN is required to supress a kernel warning for bigger allocations. since the spl_kmem_alloc_impl flag (KM_KVMEM) for kvmalloc only based allocations has been dropped its not safe to remove this line

Reply:
added a comment